From 055b850ae9b9640004997b6de0444fee065f2e21 Mon Sep 17 00:00:00 2001
From: Ladislav Smola <lsmola@redhat.com>
Date: Wed, 16 Apr 2014 14:21:32 +0200
Subject: [PATCH 1/2] Init code for Instack Undercloud

-When Undercloud was installed using Instack, the init
 code will be a little different. So for Icehouse we are
 showing both.

Change-Id: Ib2c9b74d41a912f187d2cf3c02a2e4d77682d418
---
 .../templates/overcloud/_detail_overview.html      | 78 +++++++++++++++++-----
 .../infrastructure/less/horizon_upgrades.less      |  5 ++
 .../infrastructure/less/individual_pages.less      |  7 ++
 3 files changed, 75 insertions(+), 15 deletions(-)

diff --git a/tuskar_ui/infrastructure/overcloud/templates/overcloud/_detail_overview.html b/tuskar_ui/infrastructure/overcloud/templates/overcloud/_detail_overview.html
index bcb85fb..7957972 100644
--- a/tuskar_ui/infrastructure/overcloud/templates/overcloud/_detail_overview.html
+++ b/tuskar_ui/infrastructure/overcloud/templates/overcloud/_detail_overview.html
@@ -48,17 +48,19 @@
 {% if not dashboard_urls and not overcloud.is_deploying and not overcloud.is_failed %}
 <div class="row-fluid">
   <div class="span8">
-    <p>Your OpenStack Deployment has been successfully deployed. It needs to be initialized before you will be able to use it.
-    <p/>
-
-    <button type="button" class="btn btn-primary" data-toggle="collapse" data-target="#demo">
-    Initialize
-    </button>
-    <div id="demo" class="collapse">
-      <p>You need to run the below commands on the machine where you have tripleo source code and
-         direct access through ssh to control node ({{overcloud.keystone_ip}}).
-      </p>
-      <pre>
+    {% blocktrans %}
+      Your cloud is now deployed; however, it needs to be initialized.
+    {% endblocktrans %}
+    <ul>
+      <li>
+        {% blocktrans %}
+          For <strong>Devtest</strong> users you can <a data-toggle="collapse" data-target="#devtest">see instructions here</a>.
+        {% endblocktrans %}
+        <div id="devtest" class="collapse">
+          <pre>
+  # You need to run the following commands from a machine where you have a checkout of the tripleo source code
+  # and direct access via SSH to your Overcloud control node ({{overcloud.keystone_ip}}).
+
   export TRIPLEO_ROOT=~/tripleo
   cd $TRIPLEO_ROOT
 
@@ -69,7 +71,6 @@
 
   export OVERCLOUD_ADMIN_TOKEN={{overcloud.attributes.AdminToken}}
   export OVERCLOUD_ADMIN_PASSWORD={{overcloud.attributes.AdminPassword}}
-  export OVERCLOUD_CEILOMETER_PASSWORD={{overcloud.attributes.CeilometerPassword}}
   export OVERCLOUD_CINDER_PASSWORD={{overcloud.attributes.CinderPassword}}
   export OVERCLOUD_GLANCE_PASSWORD={{overcloud.attributes.GlancePassword}}
   export OVERCLOUD_HEAT_PASSWORD={{overcloud.attributes.HeatPassword}}
@@ -96,8 +97,55 @@
       --enable-horizon \
       ${SSLBASE:+--ssl $PUBLIC_API_URL}
   keystone role-create --name heat_stack_user
-      </pre>
-    </div>
+          </pre>
+        </div>
+      </li>
+      <li>
+        {% blocktrans %}
+          For <strong>Instack</strong> users you can <a data-toggle="collapse" data-target="#instack">see instructions here</a>.
+        {% endblocktrans %}
+        <div id="instack" class="collapse">
+          <pre>
+  # You need to run the following commands from a Undercloud node where you have
+  # direct access via SSH to your Overcloud control node ({{overcloud.keystone_ip}}).
+
+  # Usually stack user has access to Overcloud control node, so switch to stack user
+  su stack
+
+  export OVERCLOUD_IP={{overcloud.keystone_ip}}
+
+  export OVERCLOUD_ADMIN_TOKEN={{overcloud.attributes.AdminToken}}
+  export OVERCLOUD_ADMIN_PASSWORD={{overcloud.attributes.AdminPassword}}
+  export OVERCLOUD_CINDER_PASSWORD={{overcloud.attributes.CinderPassword}}
+  export OVERCLOUD_GLANCE_PASSWORD={{overcloud.attributes.GlancePassword}}
+  export OVERCLOUD_HEAT_PASSWORD={{overcloud.attributes.HeatPassword}}
+  export OVERCLOUD_NEUTRON_PASSWORD={{overcloud.attributes.NeutronPassword}}
+  export OVERCLOUD_NOVA_PASSWORD={{overcloud.attributes.NovaPassword}}
+  export OVERCLOUD_SWIFT_PASSWORD={{overcloud.attributes.SwiftPassword}}
+  export OVERCLOUD_SWIFT_HASH={{overcloud.attributes.SwiftHashSuffix}}
+
+  source /etc/tripleo/overcloudrc
+
+  tripleo init-keystone -p $OVERCLOUD_ADMIN_PASSWORD $OVERCLOUD_ADMIN_TOKEN \
+      $OVERCLOUD_IP admin@example.com heat-admin@$OVERCLOUD_IP
+  tripleo setup-endpoints $OVERCLOUD_IP --cinder-password $OVERCLOUD_CINDER_PASSWORD \
+      --glance-password $OVERCLOUD_GLANCE_PASSWORD \
+      --heat-password $OVERCLOUD_HEAT_PASSWORD \
+      --neutron-password $OVERCLOUD_NEUTRON_PASSWORD \
+      --nova-password $OVERCLOUD_NOVA_PASSWORD \
+      --swift-password $OVERCLOUD_SWIFT_PASSWORD \
+      --enable-horizon
+  keystone role-create --name heat_stack_user
+          </pre>
+        </div>
+      </li>
+    </ul>
+    <p class="muted">
+      {% blocktrans %}
+        For more information, see our <a href="https://wiki.openstack.org/wiki/Tuskar/UsageGuide#Post-Deployment">documentation</a>
+        . We are working on improving your experience here.
+      {% endblocktrans %}
+    </p>
   </div>
 </div>
 {% endif %}
@@ -179,7 +227,7 @@
           </p>
         {% endfor %}
       {% else %}
-         You need to initialize your OpenStack Deployment first.
+         {% trans "You need to initialize your OpenStack Deployment first." %}
       {% endif %}
     </div>
   </div>
diff --git a/tuskar_ui/infrastructure/static/infrastructure/less/horizon_upgrades.less b/tuskar_ui/infrastructure/static/infrastructure/less/horizon_upgrades.less
index 11c3513..19f9a44 100644
--- a/tuskar_ui/infrastructure/static/infrastructure/less/horizon_upgrades.less
+++ b/tuskar_ui/infrastructure/static/infrastructure/less/horizon_upgrades.less
@@ -221,3 +221,8 @@ h1.brand a {
     margin: 0 0 2em 0;
   }
 }
+
+// general
+a[data-target] {
+  cursor: pointer;
+}
diff --git a/tuskar_ui/infrastructure/static/infrastructure/less/individual_pages.less b/tuskar_ui/infrastructure/static/infrastructure/less/individual_pages.less
index 9bc000e..e933294 100644
--- a/tuskar_ui/infrastructure/static/infrastructure/less/individual_pages.less
+++ b/tuskar_ui/infrastructure/static/infrastructure/less/individual_pages.less
@@ -67,3 +67,10 @@
     font-weight: normal;
   }
 }
+
+#detail__overview {
+  ul {
+    list-style: disc;
+    margin: 0 0 9px 25px;
+  }
+}
-- 
1.8.1.4

